### PR TITLE
docker: enables building the Zulip image with a custom CA bundle

### PR DESCRIPTION
**Pull request by xificurC**
I had a hard time building the Zulip image because we have a custom certificate in the chain. This PR along with an accompanying one in the main Zulip repository allow the user to build a docker image with a custom CA bundle. See readme update for details.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,8 +5,11 @@
 FROM ubuntu:xenial-20171114
 LABEL maintainer="Alexander Trost <galexrt@googlemail.com>"
 
-ENV ZULIP_GIT_URL="https://github.com/zulip/zulip.git" \
-    ZULIP_GIT_REF="master"
+ARG ZULIP_GIT_URL=https://github.com/zulip/zulip.git
+ARG ZULIP_GIT_REF=master
```
Comment (timabbott) [inline Dockerfile]: Can you do the refactor to turn these into arguments as a separate commit?  

Check out the Zulip commit guidelines for more details: https://zulip.readthedocs.io/en/latest/contributing/version-control.html

File: Dockerfile [hunk 2]
```diff
@@ -5,8 +5,11 @@
...
+ARG CUSTOM_CA_CERTIFICATES=
+
+SHELL ["/bin/sh", "-xc"]
```
Comment (timabbott) [inline Dockerfile]: This SHELL change should also be its own commit, with a commit message explaining why it's an improvement.

File: README.md [hunk 3]
```diff
@@ -256,6 +256,23 @@ Since that process for running management commands is a pain, we recommend
 
 [wrapper-tool]: https://github.com/zulip/docker-zulip/wiki/Running-Management-Commands
 
+### Using a custom certificate bundle
```
Comment (timabbott) [inline README.md]: Maybe add "for outgoing HTTP connection"?

File: docker-compose.yml [hunk 4]
```diff
@@ -26,6 +26,12 @@ services:
     image: "quay.io/galexrt/zulip:1.8.1-0"
     build:
       context: .
+      args:
+        # Use these if you want to download zulip from a different repo/branch
+        # ZULIP_GIT_URL:
+        # ZULIP_GIT_REF:
```
Comment (timabbott) [inline docker-compose.yml]: What do you think about including the default values here, to help make it feel easier to edit?

File: README.md [hunk 5]
```diff
@@ -256,6 +256,23 @@ Since that process for running management commands is a pain, we recommend
...
+Zulip installs packages via `yarn` and `pip` and these need to be
+configured to use these certificates. You will need to get your
+certificate bundle into the docker image, either by adding a `COPY`
+somewhere or by replacing the `FROM`s with your custom ubuntu image
+that includes your bundle. The recommended way is to have your own
+base image which has your bundle ready at the default
+`/etc/ssl/certs/ca-certificates.crt`.
+
+The next and last step is to set up the `CUSTOM_CA_CERTIFICATES`
+argument in `docker-compose.yml` to point to your CA bundle.
```
Comment (timabbott) [inline README.md]: I know this feels obvious, but we should explain what value to put there, i.e. the path to the bundle.

**Comment by xificurC** [comment]
Hi @timabbott , thanks for taking the time to go through all my PRs! All your points sound reasonable so, I'll do the splits/refactors as requested.

**Comment by xificurC** [comment]
Superseded by #147, #148 and #149
